pkg/i2gw/ingress2gateway.go

```diff
@@ -196,5 +203,24 @@ func mergeGateways(gatewaResources []GatewayResources) (map[types.NamespacedName
 		}
 	}
 
-	return newGateways, errs
+	return newGatewayContexts, errs
+}
+
+func mergeGatewayIR(current, existing *GatewayContext) {
+	current.Apisix = ir.MergeApisixGatewayIR(current.Apisix, existing.Apisix)
+	current.Gce = ir.MergeGceGatewayIR(current.Gce, existing.Gce)
+	current.IngressNginx = ir.MergeIngressNginxGatewayIR(current.IngressNginx, existing.IngressNginx)
+	current.Istio = ir.MergeIstioGatewayIR(current.Istio, existing.Istio)
+	current.Kong = ir.MergeKongGatewayIR(current.Kong, existing.Kong)
+	current.Openapi3 = ir.MergeOpenapi3GatewayIR(current.Openapi3, existing.Openapi3)
+}
```

Comment:
I am not sure I understand how all these sub-functions work, it looks like it is either returning existing, current or a null struct.